### DISCUSSION TITLE
How to get latest date?

### DISCUSSION THREAD

**Discussion by andykallioBRG** [jamessimone/apex-rollup#627] [Q&A]
Hello friends. I can't seem to get the latest date! 

I am trying to get the most recent createddate of a childobject of a particular recordtype. 

I have been careful to to make the where clause case sensitive  ```RecordTypeId = '0127F000000KK6wQAG'```

Various configurations of the custom meta data have been tried. For example, simply using the LAST operator. And have also tried the Order Bys cmdt in a few different ways. 

With each new configuration of cmdt I clear all the data from the roll up calc field and perform a full recalc using the cmdt. Regardless of the configuration of the cmdt the result is always the same...the older date is always populated. 

Thanks for any suggestions or help.

**Comment by jamessimone** [comment]
Are you setting the sort order to DESC? An order by on CreatedDate DESC with first should work

**Reply by andykallioBRG** [reply]
Hi James!  Yes, that is one of the configurations that has been tried. Here is a screenshot. I just tried another full recalc of it. And still getting the older date.  

<img width="941" alt="Screenshot 2024-09-25 130913" src="https://github.com/user-attachments/assets/06021850-2686-44e9-9099-15aeb98d2664">

**Comment by andykallioBRG** [comment]
I think I have found evidence in the debug logs that the expected cmdt configurations are being used.

`VARIABLE_SCOPE_BEGIN|[120]|meta|Rollup__mdt|true|false
16:27:19.0 (416071736)|VARIABLE_ASSIGNMENT|[120]|meta|{"IsTableFormatted__c":false,"SharingMode__c":"System","SplitConcatDelimiterOnCalcItem__c":false,"CalcItem__c":"Survey_Response__c","IsDistinct__c":true,"LookupObject__c":"Account","RollupToUltimateParent__c":false,"LookupFieldOnCalcItem__c":"Account__c","MasterLabel":"SP account latest np (1 more) ...","CalcItemWhereClause__c":"RecordTypeId = '0127 (53 more) ...","RollupOrderBys__r":[{"Rollup__c":"m0WH20000004JlAMAU","FieldName__c":"CreatedDate","Ranking__c":0.0,"SortOrder__c":"Descending","Id":"m0TH20000004Nk2MAE","NullSortOrder__c":"NULLS LAST"}],"IsFullRecordSet__c":true,"ShouldRunWithoutCustomSettingEnabled__c":false,"RollupOperation__c":"FIRST","RollupFieldOnLookupObject__c":"Latest_nps_survey_re (9 more) ...","Id":"m0WH20000004JlAMAU","RollupFieldOnCalcItem__c":"CreatedDate","LimitAmount__c":1.0,"RollupControl__c":"m0RGA000000XmpR2AS","DeveloperName":"SP_account_latest_np (1 more) ...","IsRollupStartedFromParent__c":false,"LookupFieldOnLookupObject__c":"Id"}|0x71d8b067`

**Reply by jamessimone** [reply]
Interesting. I will take a look, this should definitely be working
